src/java.base/share/classes/jdk/internal/foreign/abi/ppc64/CallArranger.java

```diff
@@ -126,15 +126,7 @@ public Bindings getBindings(MethodType mt, FunctionDescriptor cDesc, boolean for
     }
 
     public MethodHandle arrangeDowncall(MethodType mt, FunctionDescriptor cDesc, LinkerOptions options) {
-        Bindings bindings = getBindings(mt, cDesc, false, options);
-
-        MethodHandle handle = new DowncallLinker(C, bindings.callingSequence).getBoundMethodHandle();
-
-        if (bindings.isInMemoryReturn) {
-            handle = SharedUtils.adaptDowncallForIMR(handle, cDesc, bindings.callingSequence);
-        }
-
-        return handle;
+        return DowncallLinker.getBoundMethodHandle(mt, cDesc, options);
```

Comment:
This should have the same comment as other modified implementations of `arrangeDowncall()`; `arrangeUpcall()` should be modified here as well for consistency.

Reply:
Added the comments and also modified `arrangeUpcall()`.